sql/multipolygon.go

```diff
@@ -0,0 +1,264 @@
+// Copyright 2020-2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+import (
+	"reflect"
+
+	"gopkg.in/src-d/go-errors.v1"
+
+	"github.com/dolthub/vitess/go/sqltypes"
+	"github.com/dolthub/vitess/go/vt/proto/query"
+)
+
+// MultiPolygonType represents the MULTIPOLYGON type.
+// https://dev.mysql.com/doc/refman/8.0/en/gis-class-multipolygon.html
+// The type of the returned value is MultiPolygon.
+type MultiPolygonType struct {
+	SRID        uint32
+	DefinedSRID bool
+}
+
+// MultiPolygon is the value type returned from MultiPolygonType. Implements GeometryValue.
+type MultiPolygon struct {
+	SRID     uint32
+	Polygons []Polygon
+}
+
+var (
+	ErrNotMultiPolygon = errors.NewKind("value of type %T is not a multipolygon")
+
+	multipolygonValueType = reflect.TypeOf(MultiPolygon{})
+)
+
+var _ Type = MultiPolygonType{}
+var _ SpatialColumnType = MultiPolygonType{}
+var _ GeometryValue = MultiPolygon{}
+
+// Compare implements Type interface.
+// TODO: it might be better to just serialize and the compare the []byte
+func (t MultiPolygonType) Compare(a interface{}, b interface{}) (int, error) {
+	// Compare nulls
+	if hasNulls, res := compareNulls(a, b); hasNulls {
+		return res, nil
+	}
+
+	// Expect to receive a MultiPolygon, throw error otherwise
+	_a, ok := a.(MultiPolygon)
+	if !ok {
+		return 0, ErrNotMultiPolygon.New(a)
+	}
+	_b, ok := b.(MultiPolygon)
+	if !ok {
+		return 0, ErrNotMultiPolygon.New(b)
+	}
+
+	// Get shorter length
+	var n int
+	lenA := len(_a.Polygons)
+	lenB := len(_b.Polygons)
+	if lenA < lenB {
+		n = lenA
+	} else {
+		n = lenB
+	}
+
+	// Compare each polygon until there's a difference
+	for i := 0; i < n; i++ {
+		diff, err := PolygonType{}.Compare(_a.Polygons[i], _b.Polygons[i])
```

Comment:
are comparison methods not based on SRID/coordinate system?

Reply:
They should and eventually will be, but right now we don't treat the coordinate systems any differently.